core/src/test/scala/unit/kafka/message/MessageTest.scala

```diff
@@ -131,10 +131,12 @@ class MessageTest extends JUnitSuite {
     }
 
     for (v <- messages) {
-      if (v.magicValue == Message.MagicValue_V0) {
-        convertAndVerify(v, Message.MagicValue_V0, Message.MagicValue_V1)
-      } else if (v.magicValue == Message.MagicValue_V1) {
-        convertAndVerify(v, Message.MagicValue_V1, Message.MagicValue_V0)
+      if (v.codec == NoCompressionCodec) {
```

Comment:
Hmm.. I think the test was broken or at least incomplete since the `Message.toFormatVersion` only did shallow conversion. When I implemented this in the client code, I forbid shallow-only conversion because it results in bugs like what we found in `LogCleaner`. We'll probably end up dropping this code after we remove `Message.toFormatVersion` as suggested above.